File: vaso/src/domain/user_settlement.ts
import {Order} from './order';

export class UserSettlement {
    sourceOrder: Order;
    settlementStatus: string;
    fromAddress: string;
    toAddress: string;
    value: string;
    dueOnBlockchainAt: Date;

    private constructor() {
    }

    public static parse(json: any): UserSettlement {
        throw new Error();
    }
}

File: casa/src/api/user_settlement.rs
use domain::*;
use chrono::prelude::*;
use web3::types::U256;

#[derive(Serialize)]
pub struct UserSettlement {
    pub source_order: Order,
    pub settlement_status: SettlementStatus,
    pub from_address: ByteAddress,
    pub to_address: ByteAddress,
    pub value: U256,
    pub due_on_blockchain_at: DateTime<Utc>
}
